src/main/java/pl/javastart/task/logic/University.java

```diff
@@ -0,0 +1,150 @@
+    private static final int MAX_CLASSES_AMOUNT = 15;
+    private Student[] students;
+    private Lecturer[] lecturers;
+    private Group[] groups;
+
+    public University(int maxStudentsAmount, int maxLecturersAmount, int maxGroupsAmount) {
+        this.students = new Student[maxStudentsAmount];
+        this.lecturers = new Lecturer[maxLecturersAmount];
+        this.groups = new Group[maxGroupsAmount];
+    }
+
+    public Student addStudent(String firstName, String lastName, int index) {
+        Student student = new Student(firstName, lastName, index, MAX_CLASSES_AMOUNT);
+        for (int i = 0; i < students.length; i++) {
+            if (students[i] == null) {
+                students[i] = student;
+                return student;
+            }
+        }
+        return null;
+    }
+
+    public Lecturer addLecturer(String firstName, String lastName, int id, String degree) {
+        Lecturer lecturer = new Lecturer(firstName, lastName, id, degree);
+        for (int i = 0; i < lecturers.length; i++) {
+            if (lecturers[i] == null) {
+                lecturers[i] = lecturer;
+                return lecturer;
+            }
+        }
+        return null;
+    }
+
+    public Group addGroup(String code, String name, Lecturer lecturer) {
+        Group group = new Group(code, name, lecturer, MAX_STUDENTS_AMOUNT_IN_GROUP);
+        for (int i = 0; i < groups.length; i++) {
+            if (groups[i] == null) {
+                groups[i] = group;
+                return group;
+            }
+        }
+        return null;
+    }
+
+    public void addStudentToGroup(String firstName, String lastName, int index, Group group) {
+        Student student = findStudent(index);
+        if (student == null) {
+            student = addStudent(firstName, lastName, index);
+            group.addStudent(student);
+        } else if (isStudentInGroup(group, index)) {
+            System.out.printf("Student o indeksie %d jest już w grupie %s%n", index, group.getCode());
+        } else {
+            group.addStudent(student);
+        }
+    }
+
+    private boolean isStudentInGroup(Group group, int index) {
+        Student[] students = group.getStudents();
+        for (int i = 0; i < students.length; i++) {
+            if (students[i] != null && students[i].getIndex() == index) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void addGrade(Student student, Group group, double value) {
+        Grade[] grades = student.getGrades();
+        if (!doesGradeExist(grades, group.getCode())) {
+            Grade grade = new Grade(student, group, value);
+            insertGrade(grades, grade);
+        }
+    }
+
+    private void insertGrade(Grade[] grades, Grade grade) {
+        for (int i = 0; i < grades.length; i++) {
+            if (grades[i] == null) {
+                grades[i] = grade;
+                break;
+            }
+        }
```

Comment:
W tych wszystkich metodach wyszukujących też przydadzą Ci się liczniki, dzięki czemu będzie coś typu `for (int i = 0; i < gradesCounter; i++)`
bez tego ifa w środku